vendor/github.com/docker/go-connections/tlsconfig/config.go

```diff
@@ -83,6 +87,51 @@ func certPool(caFile string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
+// getPrivateKey returns the private key in 'keyBytes', in PEM-encoded format.
+// If the private key is encrypted, 'passRetriever' will be invoked to get a passphrase,
+// which is then used to decrypted the private key.
+func getPrivateKey(passRetriever notary.PassRetriever, keyBytes []byte) ([]byte, error) {
+	decryptedKey, err := trustmanager.GetDecryptedPrivateKey(passRetriever, keyBytes, "private_key", "")
+	if err != nil {
+		return nil, err
+	}
+
+	return utils.KeyToPEM(decryptedKey, "")
+}
+
+// getCert returns a Certificate from the CertFile and KeyFile in 'options',
+// if the key is encrypted, the PassphraseRetriever in 'options' will be invoked
+// to get a passphrase.
+func getCert(options Options) (tls.Certificate, error) {
+	if options.CertFile == "" && options.KeyFile == "" {
+		return tls.Certificate{}, nil
+	}
+
+	errTemplate := "Could not load X509 key pair: %v."
+
+	cert, err := ioutil.ReadFile(options.CertFile)
+	if err != nil {
+		return tls.Certificate{}, fmt.Errorf(errTemplate, err)
+	}
+	prKeyBytes, err := ioutil.ReadFile(options.KeyFile)
+	if err != nil {
+		return tls.Certificate{}, fmt.Errorf(errTemplate, err)
+	}
+
+	if options.PassphraseRetriever != nil {
```

Comment:
Rather than get the passphrase retriever here, since the TLS key must be PEM encoded (`tls.LoadX509KeyPair` also requires PEM encoded files), maybe we can just decode (using `pem.Decode`) the contents, determine if it's encrypted (`x509.IsEncryptedPEMBlock`), and if it is attempt to decrypt it using the passphrase.  If that fails, return an error that indicates that the key was encrypted.  If it succeeds, then we have the key. 

That way the caller (docker in this case) could determine whether or not to fail, because the key was just malformed garbage, or to attempt to use a passphrase retriever because the key is encrypted.